desc/objectives/_geometry.py

```diff
@@ -459,6 +459,81 @@ def compute(self, params, constants=None):
         return data["V"]
 
 
+class CoilLength(_Objective):
+    """Coil length."""
+
+    def __init__(
+        self,
+        coil,
+        target=2 * np.pi,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        loss_function=None,
+        deriv_mode="auto",
+        name="coil-length",
+    ):
+        super().__init__(
+            things=coil,
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            loss_function=loss_function,
+            deriv_mode=deriv_mode,
+            name=name,
+        )
+
+    def build(self, use_jit=True, verbose=1):
+        """Build objective function."""
+        coil = self.things[0]
+        self._dim_f = 1
+        # TODO: is this the correct data key?
+        # seems like it because it gives the circumference of a circular coil?
+        self._data_keys = ["length"]
+
+        timer = Timer()
+        if verbose > 0:
+            print("Precomputing transforms")
+        timer.start("Precomputing transforms")
+
+        grid = LinearGrid(L=4, M=4)
+        # TODO: what grid is supposed to be used here?
+        # TODO: am I supposed to use obj=coil?
+        profiles = get_profiles(self._data_keys, obj=coil, grid=grid)
+        transforms = get_transforms(self._data_keys, obj=coil, grid=grid)
+        self._constants = {
+            "transforms": transforms,
+            "profiles": profiles,
+        }
+
+        timer.stop("Precomputing transforms")
+        if verbose > 1:
+            timer.disp("Precomputing transforms")
+
+        super().build(use_jit=use_jit, verbose=verbose)
+
+    def compute(self, params, constants=None):
+        """Compute length of coil."""
+        if constants is None:
+            constants = self._constants
+
+        # TODO: gets KeyError with 'rotmat' when computing 'x_s'.
```

Comment:
if you get this when using the objective function, this is fixed  in #840 , try merging that one in and see if it works